radicle/src/profile.rs

```diff
@@ -250,6 +250,13 @@ mod test {
                 .join("Radicle"),
         )
         .unwrap();
-        assert_eq!(home.path, path);
+        // Handle MacOS test case with '/private' tmp.path prefix.
+        let home_path = if let Ok(path) = home.path.strip_prefix("/private") {
+            ("/".to_owned() + path.to_str().unwrap()).into()
+        } else {
+            home.path
+        };
```

Comment:
I'm not 100% sure if this code will work but we could definitely use parts of it if it doesn't.
I added a check for the OS being `macos` since it only affects that OS, so it's a good signal for testing only on that platform.
Secondly, I opted to use `Path::new("/").join(path)` instead of string manipulation.
Last, I think if we could mark `home` as `mut` then perhaps we can just assign the new `home.path`. If this doesn't work we can use the new `home_path` variable and have an `#[cfg(not(target_os = "macos"))]` branch of execution and do `home_path = home.path`.
```suggestion
        #[cfg(target_os = "macos")]
        // Handle MacOS test case with '/private' tmp.path prefix.
        if let Ok(path) = home.path.strip_prefix("/private") {
            home.path = Path::new("/").join(path);
        };
```

Reply:
Thanks @FintanH!

I also tried a few of your comments, like mutate `home` and add platform specific branch execution and mutate the variable inside of a platform specific tag didn't worked yesterday, but today it seems like the compiler says yes 😅

Reply:
Unfortunately, the linter says no :joy: I think your original proposal is fine with the `if ... else ...` but let's add the `join`.

Reply:
hmm it seems so, ok will revert so the linter says yes